### PR TITLE
[prometheus-node-exporter] Add a pod monitor (#2861)

### LINKED ISSUE #2861: [prometheus-node-exporter] enable podMonitor for node-exporter
### Is your feature request related to a problem ?

Hey, I would like to use `podMonitor` instead of `serviceMonitor` with the node-exporter, today we have a problem with the serviceMonitor of the node exporter. the reason is that in some of our environments we have more than 1000 nodes and which creates more than 1000 endpoints for the node-exporter service and there is a limit of 1000 endpoints per service. that limit causes us to lose data from the node-exporter.
The chart has built-in support for serviceMonitor in the values and I would like the podMonitor to have the same.

Until this one going to be merged - https://github.com/prometheus-operator/prometheus-operator/issues/3862

### Describe the solution you'd like.

built-in option in the value to add podMonitor

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by zeritti**
Signed-off-by: zeritti <47476160+zeritti@users.noreply.github.com>

#### What this PR does / why we need it
This PR brings an option do define a podmonitor for Node Exporter pods.

#### Which issue this PR fixes

- fixes #2861

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by zeritti** [comment]
Thank you @zanhsieh will do.

**Review by zanhsieh — APPROVED** [review]

File: charts/prometheus-node-exporter/README.md [hunk 1]
```diff
@@ -41,6 +41,12 @@ helm upgrade [RELEASE_NAME] [CHART] --install
 
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
+### 4.8.x to 4.9.0
```
Comment (monotek) [inline charts/prometheus-node-exporter/README.md:44]: i don´t think upgrading instructions are needed, as this is no major version change with breaking changes.
instead add this info above the podmonitor config please.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/README.md (hunk 1 above)
Comment (zeritti) [reply charts/prometheus-node-exporter/README.md]: Upgrade note removed, comment added to the configuration instead.

**Review by monotek — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
please fix merge conflicts

**Comment by zeritti** [comment]
Merge conflict cleared.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**